MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m

```diff
@@ -208,4 +216,62 @@ - (void)sessionTracker:(id)sessionTracker processLog:(id<MSLog>)log withPriority
   [self sendLog:log withPriority:priority];
 }
 
+
++ (void)setDelegate:(_Nullable id <MSAnalyticsDelegate>)delegate {
+  [[self sharedInstance] setDelegate:delegate];
+}
+
+#pragma mark - MSChannelDelegate
+
+- (void)channel:(id)channel willSendLog:(id<MSLog>)log {
+  if (!self.delegate) {
+    return;
+  }
+  NSObject *logObject = (NSObject *)log;
+  if ([logObject isKindOfClass:[MSEventLog class]] &&
+      [self.delegate respondsToSelector:@selector(analytics:willSendEventLog:)]) {
+    MSEventLog *eventLog = (MSEventLog*)log;
+    [self.delegate analytics:self willSendEventLog:eventLog];
+  }
+  else if ([logObject isKindOfClass:[MSPageLog class]] &&
+           [self.delegate respondsToSelector:@selector(analytics:willSendPageLog:)]) {
+    MSPageLog *pageLog = (MSPageLog*)log;
+    [self.delegate analytics:self willSendPageLog:pageLog];
+  }
+}
+
+- (void)channel:(id<MSChannel>)channel didSucceedSendingLog:(id<MSLog>)log {
+  if (!self.delegate) {
+    return;
+  }
+  NSObject *logObject = (NSObject *)log;
+  if ([logObject isKindOfClass:[MSEventLog class]] &&
+      [self.delegate respondsToSelector:@selector(analytics:didSucceedSendingEventLog:)]) {
```

Comment:
@jaelim-ms I just figured that since they are different types, they could be handled differently. And would save any client code from having to deal with MSLogWithProperties and casting. I don't really have any strong opinions on the matter though. (The interface is not public, anyways. It's only going to be used by Xamarin UITests.)